tests/test_selenium.py

```diff
@@ -0,0 +1,144 @@
+    driver = create_driver
+    e_bike_page = MainPage(driver).geo_popup_check().accept_cookie().open_electric_bikes()
+
+    if e_bike_category == 'mountain':
+        e_bike_page.open_category_electric_mountain_bikes().geo_popup_check().accept_cookie()
+        assert e_bike_page.check_el_mnt_bike_page()
+    elif e_bike_category == 'road':
+        e_bike_page.open_category_electric_road_bikes()
+        assert e_bike_page.check_el_rd_bike_page()
+    elif e_bike_category == 'electra':
+        e_bike_page.open_category_electra_e_bikes()
+        assert e_bike_page.check_el_e_bike_page()
+    elif e_bike_category == 'hybrid':
+        e_bike_page.open_category_electric_hybrid_bikes()
+        assert e_bike_page.check_el_hb_bike_page()
+
+
+@pytest.mark.smoke
+@pytest.mark.regression
+@pytest.mark.parametrize("e_bike_category, sorting_by", [("mountain", "price-desc"),
+                                                         ("mountain", "price-asc"),
+                                                         ("road", "price-desc"),
+                                                         ("road", "price-asc"),
+                                                         ("electra", "price-desc"),
+                                                         ("electra", "price-asc"),
+                                                         ("hybrid", "price-desc"),
+                                                         ("hybrid", "price-asc")
+                                                         ])
+def test_open_the_most_expensive_or_cheapest_e_bike(create_driver, e_bike_category, sorting_by):
+    driver = create_driver
+    e_bike_page = MainPage(driver).geo_popup_check().accept_cookie().open_electric_bikes()
+
+    if e_bike_category == 'mountain':
+        e_bike_page.open_category_electric_mountain_bikes().geo_popup_check()
+        Select(driver.find_element(by='id', value='sortOptions1')).select_by_value(sorting_by)
+        ElectricBikesPage(driver).open_first_bike_in_list().select_biggest_bike().add_bike_to_cart()
+    elif e_bike_category == 'road':
+        e_bike_page.open_category_electric_road_bikes().geo_popup_check()
+        Select(driver.find_element(by='id', value='sortOptions1')).select_by_value(sorting_by)
+        ElectricBikesPage(driver).open_first_bike_in_list().select_biggest_bike().add_bike_to_cart()
+    elif e_bike_category == 'electra':
+        e_bike_page.open_category_electra_e_bikes().geo_popup_check()
+        Select(driver.find_element(by='id', value='sortOptions1')).select_by_value(sorting_by)
+        ElectricBikesPage(driver).open_first_bike_in_list().select_biggest_bike().add_bike_to_cart()
+    elif e_bike_category == 'hybrid':
+        e_bike_page.open_category_electric_hybrid_bikes().geo_popup_check()
+        Select(driver.find_element(by='id', value='sortOptions1')).select_by_value(sorting_by)
+        ElectricBikesPage(driver).open_first_bike_in_list().select_biggest_bike().add_bike_to_cart()
```

Comment:
it is not a test at all. where is asserts? 


General comment. Why all TC are in one file? please create different suites. 
A lot of code duplication. Please think about fictures for you code